src/reconcile/rec_write.c

```diff
@@ -2970,9 +2971,10 @@ __rec_write_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
         mod->rec_result = WT_PM_REC_REPLACE;
         break;
     default: /* Page split */
-        if (WT_PAGE_IS_INTERNAL(page))
+        if (WT_PAGE_IS_INTERNAL(page)) {
```

Comment:
like previous comment, here we should do for all cases not only for internal page.